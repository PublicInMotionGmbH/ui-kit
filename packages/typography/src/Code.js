import React from 'react'
import PropTypes from 'prop-types'

import { buildClassName } from '@talixo/shared'

const propTypes = {
  /** Additional class name */
  className: PropTypes.string,

  /** Code which should be shown there */
  children: PropTypes.node,

  /** Code language, ignored for now */
  language: PropTypes.string
}

/**
 * Component which represents code in typography.
 * TODO: Make possibility of syntax highlighting with optional dependency.
 *
 * @param {object} props
 * @param {string} [props.className]
 * @param {*} [props.children]
 * @param {string|*} [props.secondary]
 * @returns {React.Element}
 */
function Code (props) {
  const { className, children, ...passedProps } = props

  return (
    <code className={buildClassName('code', className)} {...passedProps}>
      {children}
    </code>
  )
}

Code.propTypes = propTypes

export default Code
