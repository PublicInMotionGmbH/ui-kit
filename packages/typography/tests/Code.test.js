import React from 'react'
import { shallow } from 'enzyme'

import Code from '../src/Code'

describe('<Code />', () => {
  it('renders empty correctly', () => {
    const wrapper = shallow(<Code />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders children correctly', () => {
    const wrapper = shallow(
      <Code>
        const x = 10
      </Code>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
