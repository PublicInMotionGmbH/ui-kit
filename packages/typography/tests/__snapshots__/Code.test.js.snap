// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> renders children correctly 1`] = `
<code
  className="talixo-code"
>
  const x = 10
</code>
`;

exports[`<Code /> renders empty correctly 1`] = `
<code
  className="talixo-code"
/>
`;
